diff --git a/client/src/imgproc.rs b/client/src/imgproc.rs
index f1a6941..3d9f419 100644
--- a/client/src/imgproc.rs
+++ b/client/src/imgproc.rs
@@ -150,9 +150,10 @@ pub struct VectorImage<'a>(&'a Tree);
 impl RasterImage<'_> {
     pub fn decode(&self, format: PixelFormat) -> Result<Image, String> {
         let imgbuf = self.0;
-        let reader = image::ImageReader::new(Cursor::new(&imgbuf.bytes))
+        let mut reader = image::ImageReader::new(Cursor::new(&imgbuf.bytes))
             .with_guessed_format()
             .map_err(|e| format!("failed to read image: {e}"))?;
+        reader.no_limits();
         let dynimage = reader
             .decode()
             .map_err(|e| format!("failed to decode image: {e}"))?;

